tests/unit/network/probe/ProbeServer.spec.js

```diff
@@ -0,0 +1,105 @@
+const assert = require('assert');
+const { ProbeServer } = require('../../../../lib/network/probe/ProbeServer');
+const http = require('http');
+
+function makeRequest(method, uri, cb) {
+    const params = {
+        hostname: 'localhost',
+        port: 4042,
+        path: uri,
+        method,
+    };
+    const req = http.request(params);
+    req.setNoDelay(true);
+    req.on('response', res => {
+        cb(undefined, res);
+    });
+    req.on('error', err => {
+        assert.ifError(err);
+        cb(err);
+    }).end();
+}
+
+describe('network.probe.ProbeServer', () => {
+    /** @type {ProbeServer} */
+    let server;
+
+    beforeEach(done => {
+        server = new ProbeServer({ port: 4042 });
+        server._cbOnListening = done;
```

Comment:
what does this do?

Reply:
I think this is a callback used for when the server is up and listening, I believe